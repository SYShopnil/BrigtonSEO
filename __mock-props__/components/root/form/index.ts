// Internal Importstypes
import {
  IForm,
  FormElementEnum,
  ButtonColorSchemeEnum,
  ButtonSizeEnum,
} from '@src/types/root';

// Internal formData
export const formData: IForm = {
  formTitle: 'Send Us Message',
  formFields: [
    {
      elementType: FormElementEnum.Input,
      type: 'text',
      name: 'firstName',
      placeholder: 'First Name',
      isRequired: true,
      space: 0.5,
    },
    {
      elementType: FormElementEnum.Input,
      type: 'text',
      name: 'lastName',
      placeholder: 'Last Name',
      isRequired: true,
      space: 0.5,
    },
    {
      elementType: FormElementEnum.Input,
      type: 'text',
      name: 'email',
      placeholder: 'Enter Email Address',
      isRequired: true,
      space: 1,
    },
    {
      elementType: FormElementEnum.Input,
      type: 'text',
      name: 'phone',
      placeholder: 'Phone number',
      isRequired: false,
      space: 1,
    },
    {
      elementType: FormElementEnum.TextArea,
      name: 'message',
      placeholder: 'Write Your Message Here',
      isRequired: true,
      space: 1,
      type: 'text',
    },
    {
      label: 'Terms & Condition',
      type: 'checkbox',
      elementType: FormElementEnum.Checkbox,
      name: 'terms',
      isRequired: true,
      space: 1,
    },
  ],
  buttonInfo: {
    buttonText: 'Send Now',
    colorScheme: ButtonColorSchemeEnum.SkyblueBgTextWhite,
    hoverColorScheme: ButtonColorSchemeEnum.SkyblueVerticalGradientBgTextWhite,
    size: ButtonSizeEnum.Small,
    'data-testid': 'submit-button',
  },
  onSubmit: () => {},
};
