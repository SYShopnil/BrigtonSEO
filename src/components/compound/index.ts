import { ContactSection } from './contact-section';
import { ListContentSection } from './list-content-section';
import BuyTicketCard from './card/buy-ticket';
import { HeroGeneral, HeroHome } from './hero-section';
import CourseCard from './card/course';
import PricingCard from './card/pricing';
import { Testimonial } from './card/testimonial';
import Person from './card/person';
import UpcomingCard from './card/upcoming';
import FooterSection from './footer-section';
import { TestimonialSection } from './testimonial-section';
import { EventHistoryCard } from './card/event-history';
import { TextAccordion } from './textAccordion';
import { ImageAccordion } from './image-accordion';

export {
  ContactSection,
  CourseCard,
  FooterSection,
  ListContentSection,
  Testimonial,
  HeroGeneral,
  BuyTicketCard,
  PricingCard,
  Person,
  EventHistoryCard,
  UpcomingCard,
  TestimonialSection,
  HeroHome,
  TextAccordion,
  ImageAccordion,
};
