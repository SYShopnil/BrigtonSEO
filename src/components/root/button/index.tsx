import Button from './button';
import { CtaButton } from './cta-button';

export { Button, CtaButton };
