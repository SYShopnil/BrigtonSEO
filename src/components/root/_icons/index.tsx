/**
 * External Imports
 */
import {
  FiCornerUpRight,
  FiCornerUpLeft,
  FiSend,
  FiNavigation,
  FiCalendar,
} from 'react-icons/fi';
import {
  RiTrainFill,
  RiFlightTakeoffFill,
  RiBusFill,
  RiMotorbikeLine,
  RiArrowRightLine,
  RiArrowLeftLine,
  RiFacebookFill,
  RiTwitterFill,
  RiInstagramFill,
  RiYoutubeFill,
  RiCalendarEventFill,
  RiMapPinFill,
} from 'react-icons/ri';
import { GoPlus } from 'react-icons/go';
import { AiOutlineMinusCircle, AiOutlinePlusCircle } from 'react-icons/ai';
import { FaUserAlt } from 'react-icons/fa';

import DoubleQuotation from './double-quotation';

/**
 * Internal Imports
 */
import { IconEnum } from '@src/types/root';
import { FooterCustomRocket } from './footer-custom-rocket';
import { Logo } from './logo';
import { EventInfoCircle } from './event-info-circle';

/**
 * @method iconSelector
 * @param {iconName} IconEnum
 * @returns {JSX.Element}
 * @description this function will receive iconName and return icon JSX.Element(svg)
 */

interface IconStoreProps {
  iconName: IconEnum;
}

const IconStore = ({ iconName }: IconStoreProps): JSX.Element => {
  switch (iconName) {
    case IconEnum.CornerUpRight: {
      return <FiCornerUpRight />;
    }
    case IconEnum.CornerUpLeft: {
      return <FiCornerUpLeft />;
    }
    case IconEnum.Send: {
      return <FiSend />;
    }
    case IconEnum.Navigation: {
      return <FiNavigation />;
    }
    case IconEnum.Calendar: {
      return <FiCalendar />;
    }
    case IconEnum.TrainFill: {
      return <RiTrainFill />;
    }
    case IconEnum.FlightTakeoffFill: {
      return <RiFlightTakeoffFill />;
    }
    case IconEnum.BusFill: {
      return <RiBusFill />;
    }
    case IconEnum.MotorbikeLine: {
      return <RiMotorbikeLine />;
    }
    case IconEnum.ArrowRightLine: {
      return <RiArrowRightLine />;
    }
    case IconEnum.ArrowLeftLine: {
      return <RiArrowLeftLine />;
    }
    case IconEnum.DoubleQuotation: {
      return <DoubleQuotation />;
    }
    case IconEnum.Facebook: {
      return <RiFacebookFill />;
    }
    case IconEnum.Twitter: {
      return <RiTwitterFill />;
    }
    case IconEnum.Instagram: {
      return <RiInstagramFill />;
    }
    case IconEnum.Youtube: {
      return <RiYoutubeFill />;
    }
    case IconEnum.FooterCustomRocket: {
      return <FooterCustomRocket />;
    }
    case IconEnum.Logo: {
      return <Logo />;
    }
    case IconEnum.CalendarEventFill: {
      return <RiCalendarEventFill />;
    }
    case IconEnum.MapPinFill: {
      return <RiMapPinFill />;
    }
    case IconEnum.EventInfoCircle: {
      return <EventInfoCircle />;
    }
    case IconEnum.AiOutlinePlusCircle: {
      return <AiOutlinePlusCircle />;
    }
    case IconEnum.AiOutlineMinusCircle: {
      return <AiOutlineMinusCircle />;
    }
    case IconEnum.FilledPlus: {
      return <GoPlus />;
    }
    case IconEnum.UserAlt: {
      return <FaUserAlt />;
    }
  }
};

export default IconStore;
