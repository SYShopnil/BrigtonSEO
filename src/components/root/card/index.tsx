/**
 * Internal Imports
 */
import { ICard } from '@src/types/root';
import Styles from './card.module.scss';

/**
 * Props
 */
interface Props extends ICard {
  children: JSX.Element;
}

/**
 *
 * @param {object}  Props
 * @returns JSX.Element
 * @description This component is a root card component which take a children component and render those here.
 */
export const Card = ({
  children,
  dataTestId,
  cardBorderRadius,
  cardShadow,
  hoverCardShadow,
}: Props) => {
  return (
    <div
      className={`relative overflow-hidden bg-white ${Styles['card']} ${
        Styles[cardBorderRadius]
      } ${Styles[cardShadow]} ${
        hoverCardShadow ? Styles[`hover-${hoverCardShadow}`] : ''
      }`}
      data-testid={`${dataTestId}`}
    >
      {children}
    </div>
  );
};
