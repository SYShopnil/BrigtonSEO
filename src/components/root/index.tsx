import { Button, CtaButton } from './button';
import { SectionHeading } from './section-heading';
import { Form } from './form';
import { Card } from './card';
import { default as Swiper, SwiperItem } from './swiper';

export { Button, SectionHeading, Form, CtaButton, Card, Swiper, SwiperItem };
