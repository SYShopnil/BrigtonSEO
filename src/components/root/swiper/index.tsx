import React, { useState } from 'react';
import SwiperCore, { Autoplay } from 'swiper';
import { Swiper } from 'swiper/react';
import 'swiper/swiper-bundle.min.css';
export { SwiperSlide as SwiperItem } from 'swiper/react';
import { useIsDomReady, useViewport, useOffset } from '@src/hooks';
import { ISwiperProps, IBreakpoints } from '@src/types/root';
import Styles from './index.module.scss';

// Inject Autoplay
SwiperCore.use([Autoplay]);

/**
 * default values
 */
const AUTOPLAY_DELAY = 4; // in seconds;
const SPACE_BETWEEN: IBreakpoints = {
  sm: 12,
  md: 16,
  lg: 20,
  xl: 24,
  '2xl': 24,
};
const SHOW_ITEMS: IBreakpoints = {
  sm: 1.2,
  md: 2,
  lg: 3,
  xl: 4,
  '2xl': 4,
};

/**
 * @param {object} ISwiperProps
 * @returns JSX.Element | null
 */
const Carousel = ({
  autoItemCount = false,
  children,
  isLoop = false,
  isCenter = false,
  autoPlay = false,
  pauseOnHover = false,
  haveOffset = false,
  spaceBetween = SPACE_BETWEEN,
  showItems = SHOW_ITEMS,
  className = '',
  speed = 800,
}: ISwiperProps): JSX.Element | null => {
  const [isDraggging, setIsDragging] = useState<boolean>(false);

  // custom hooks
  const { container } = useViewport();
  const offset = useOffset();

  /**
   * Used for - react hydration.
   * https://nextjs.org/docs/messages/react-hydration-error
   */
  const isDomReady = useIsDomReady();

  // autoplay
  const autoplaySettings = {
    delay: AUTOPLAY_DELAY * 1000,
    disableOnInteraction: false,
    pauseOnMouseEnter: pauseOnHover,
  };

  // check isDomReady
  if (!isDomReady) return null;

  // style implement for offset layout
  let style = {};
  if (haveOffset) style = { paddingLeft: offset, paddingRight: offset };

  /**
   * Tailwind & Module Classes
   */

  // swiperMainWrapper
  const swiperMainWrapper = `swiper-main-wrapper ${
    Styles['swiper-main-wrapper']
  } ${isDraggging ? Styles['cursor-dragging'] : Styles['cursor-drag']}`;
  // swiperContainer
  const swiperContainer = `swiper-main-container ${className}`;

  return (
    <div className={swiperMainWrapper}>
      <Swiper
        style={style}
        speed={speed}
        loop={isLoop}
        autoplay={autoPlay && autoplaySettings}
        centeredSlides={isCenter}
        slidesPerView={autoItemCount ? 'auto' : showItems[container]}
        spaceBetween={spaceBetween[container]}
        onTouchStart={() => {
          setIsDragging(true);
        }}
        onTouchEnd={() => {
          setIsDragging(false);
        }}
        className={swiperContainer}
      >
        {children}
      </Swiper>
    </div>
  );
};

export default Carousel;
