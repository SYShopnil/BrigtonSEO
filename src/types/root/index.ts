import {
  ButtonColorSchemeEnum,
  ButtonSizeEnum,
  IRedirectButton,
  ICtaButton,
  IButton,
  CtaButtonColorSchemeEnum,
  CtaButtonVariantEnum
} from './button';

import {
  ISectionHeading,
  SectionHeadingColorschemesEnum,
} from './section-heading';

import {
  IFormField,
  IFormValues,
  IFormFieldProp,
  IForm,
  FormElementEnum,
} from './form';

import { IconEnum } from './icon';
import { ICard, CardShadowEnum, CardBorderRadiusEnum } from './card';
import { ISwiperProps, IBreakpoints } from './swiper';

//export part

export type {
  IRedirectButton,
  ICtaButton,
  ISectionHeading,
  IFormField,
  IFormValues,
  IFormFieldProp,
  IForm,
  IButton,
  ICard,
  ISwiperProps,
  IBreakpoints,
};

export {
  FormElementEnum,
  ButtonColorSchemeEnum,
  ButtonSizeEnum,
  IconEnum,
  SectionHeadingColorschemesEnum,
  CardShadowEnum,
  CardBorderRadiusEnum,
  CtaButtonColorSchemeEnum,
  CtaButtonVariantEnum
};
