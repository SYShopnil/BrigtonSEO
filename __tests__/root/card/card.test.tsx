import { Card } from '@src/components/root';
import { CardShadowEnum } from '@src/types/root';
import { render } from '@testing-library/react';
import {
  childElementTestId,
  commonProps,
  ICommonProps,
  testBoxShadowData,
} from './config';

/**
 *
 * Demo element
 *
 */
const children = () => {
  return (
    <div
      className={`flex justify-between p-[25px] `}
      data-testid={childElementTestId}
    >
      <div>
        <h1>Right Side part</h1>
      </div>
      <div>
        <h1>Left Side part</h1>
      </div>
    </div>
  );
};

describe('When card element is called', () => {
  it('expect card to render correctly', () => {
    const { getByTestId } = render(<Card {...commonProps}>{children()}</Card>);
    const card = getByTestId(commonProps.dataTestId);
    expect(card).toBeInTheDocument();
    expect(card).toMatchSnapshot();
  });
});

/**
 *
 * Test box-shadow functionality
 *
 **/
describe.each(testBoxShadowData)(
  'When shadow props are passed',
  (shadowClass) => {
    it('expect card to have appropriate shadow in its class name', () => {
      const props: ICommonProps = {
        ...commonProps,
        cardShadow: shadowClass,
      };
      const { getByTestId } = render(<Card {...props}>{children()}</Card>);
      const cardWrapper = getByTestId(commonProps.dataTestId);
      expect(cardWrapper).toHaveClass(shadowClass);
    });
  }
);

/**
 *
 * Check whether the child element is being rendered
 *
 */
describe('When a child element is passed', () => {
  it('expect child to exist in document', () => {
    const props: ICommonProps = {
      ...commonProps,
      cardShadow: CardShadowEnum.ShadowOne,
    };

    const { queryByTestId } = render(<Card {...props}>{children()}</Card>);
    const childElement = queryByTestId(childElementTestId);
    expect(childElement).toBeInTheDocument();
  });
});
