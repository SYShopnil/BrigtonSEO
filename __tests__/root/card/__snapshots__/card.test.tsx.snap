// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When card element is called expect card to render correctly 1`] = `
<div
  class="relative overflow-hidden bg-white card radius-2xl shadow-One hover-shadow-none"
  data-testid="mainWrapper"
>
  <div
    class="flex justify-between p-[25px] "
    data-testid="childElementWrapper"
  >
    <div>
      <h1>
        Right Side part
      </h1>
    </div>
    <div>
      <h1>
        Left Side part
      </h1>
    </div>
  </div>
</div>
`;
